jni/util/vtx_manager.h

```diff
@@ -0,0 +1,312 @@
+#include <time.h>
+
+
+typedef struct GlassRacingChnlUI GlassRacingChnlUI, *PGlassRacingChnlUI;
+typedef struct GlassRacingChnlUI GlassRacingChnlUI_t;
+
+struct GlassRacingChnlUI {
+	char dummy[0x4a0];
+	float gs_voltage;
+	char dummy2[0xa];
+	uint16_t gs_bitrate;
+	char dummy3[0x200]; //ends at 0x6b0
+
+}__attribute__((packed, aligned(1)));
+
+static GlassRacingChnlUI *chnl_ui;
+
+typedef enum gs_link_stat_t {
+    GS_LINK_STAT_NORMAL=0,
+    GS_LINK_STAT_WEAK=1,
+    GS_LINK_STAT_LOST=2,
+    GS_LINK_STAT_UKNOWN=3
+} gs_link_stat_t;
+
+typedef enum Record_sender {
+    RECORD_BUTTON=0,
+    RECORD_DISARM=1
+} Record_sender;
+
+typedef enum record_state {
+    RECORD_STATE_IDLE=0,
+    RECORD_STATE_RECORDING=1,
+    RECORD_STATE_STOP_ERROR=2,
+    RECORD_STATE_STOP_FULL=3,
+    RECORD_STATE_STOP_WRITE_SLOW=4,
+    RECORD_STATE_STARTING=5,
+    RECORD_STATE_STOPPING=6
+} record_state;
+
+typedef enum record_state record_state_t;
+
+typedef struct __gs_queue __gs_queue, *P__gs_queue;
+typedef struct __gs_queue gs_queue_t;
+
+struct __gs_queue {
+    uint32_t * addr;
+    uint32_t size;
+    uint32_t num;
+    uint32_t writer;
+    uint32_t reader;
+};
+
+typedef struct __gs_gui_config __gs_gui_config, *P__gs_gui_config;
+typedef struct __gs_gui_config gs_gui_config_t;
+
+struct __gs_gui_config {
```

Comment:
we should probably not re-define __gs_gui_config as it's already defined in https://github.com/fpv-wtf/msp-osd/blob/f9d017bf6a39d07a1e96a33220cfbf168e03aebb/jni/rec/rec_shim.h#L5585.

Best would be to move the common parts to a common gs_gui_shim.c and include that.

Reply:
I was already about the double declaration of some times but some other types for some declarations have different types on both files, but I will revisit this to about the duplication as I also agree that we should not have it like that.

Reply:
Duplicated definition removed :)